### PR TITLE
hide alerts fixing #4287

### PR DESCRIPTION
**Pull request by albertlast**
My try to solve the issue #4287
possible issue is here when you got a limit of 10 and
the last 10 are hide then you didn't see the 11th.

it's not the clean version which i like to do,
but i didn't got a clear vision how to create better version.

### CODE REVIEW

File: Sources/Profile-View.php [hunk 1]
```diff
@@ -324,11 +324,20 @@ function fetch_alerts($memID, $all = false, $counter = 0, $pagination = array())
 		if (!empty($alert['text']))
 			continue;
 		if (isset($alert['extra']['board']))
-			$alerts[$id_alert]['extra']['board_msg'] = $boards[$alert['extra']['board']];
+			if ($boards[$alert['extra']['board']] == $txt['board_na'])
+				unset($alerts[$id_alert]);
+			else
+				$alerts[$id_alert]['extra']['board_msg'] = $boards[$alert['extra']['board']];
```
Comment (jdarwood007) [inline Sources/Profile-View.php]: I'm only looking at the code here and not the context, but why are we checking this against $txt?

Comment (albertlast) [reply Sources/Profile-View.php]: all alerts where intialize with the txt https://github.com/albertlast/SMF2.1/blob/82522334aa8ee193ba505ed2a8c354be1ce8259e/Sources/Profile-View.php#L263
and i placed the check here if the intial text is in here or changed.
When the array got the intial value, the user doesn't hav the access right -> remove this from the list.

**Comment by frandominguezl** [comment]
This is the result after applying the fix;

File: Sources/Profile-View.php (hunk 1 above)
Comment (jdarwood007) [reply Sources/Profile-View.php]: Ok, seems weird we are doing the N/A, but ok.

**Comment by albertlast** [comment]
well the last pr fix the issue that the numbers of new alerts showing is wrong.

what it do,
everytime a new alert come in,
the alerts field it's empty (null).
When then the first read action happen on alerts i look if the field is empty,
when not all is fine,
when empty(null) than i calc the amount.

but to be honest it's not feeling right.

alternativ:
I could place the count logic into updateMemberData but then i need to know the {query_see_board} of the receiving user, this could evade us from changing the table designe because keep not null able...

**Comment by albertlast** [comment]
I'm very happy to bring a new solution of counting alerts.

first i revert the last two changes and added a new way,
this way do three things.

First it add a new function build_query_board($userid) to get the {query_see_board} from not current users.
Secondly it's fix fetch_alert where you can provide with $memID but this is calc with rights of the current user -> now it's calc with the provided memID.
Third updateMemberData calc alerts now on the query_see_board of every different user.

**Review by Idanban12 — APPROVED** [review]

**Pull request merged**